File: src/main/java/phss/wsworkcars/controllers/MarcaController.java
package phss.wsworkcars.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import phss.wsworkcars.models.Marca;
import phss.wsworkcars.services.MarcaService;

import java.util.List;

@RestController
public class MarcaController {

    @Autowired
    private MarcaService service;

    @GetMapping("/marcas")
    public List<Marca> getMarcas() {
        return service.getData();
    }

    @GetMapping("/getMarca/{id}")
    public Marca getMarcaById(@PathVariable long id) {
        return service.getDataById(id).orElse(null);
    }

    @GetMapping("/getMarcaByName/{name}")
    public Marca getMarcaByName(@PathVariable String name) {
        return service.getMarcaByName(name).orElse(null);
    }

    @PostMapping("/addMarca")
    public Marca addMarca(@RequestBody Marca marca) {
        return service.saveData(marca);
    }

    @PostMapping("/addMarcas")
    public List<Marca> addMarcas(@RequestBody List<Marca> marcas) {
        return service.saveDataList(marcas);
    }

    @PutMapping("/updateMarca")
    public Marca updateMarca(@RequestBody Marca marca) {
        return service.updateMarca(marca);
    }

    @DeleteMapping("/deleteMarca/{id}")
    public HttpStatus deleteMarca(@PathVariable long id) {
        Marca result = service.deleteDataById(id);

        if (result == null) return HttpStatus.NOT_FOUND;
        else return HttpStatus.OK;
    }

}

File: src/main/kotlin/phss/wsworkcars/controllers/MarcaController.kt
package phss.wsworkcars.controllers

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import phss.wsworkcars.models.Marca
import phss.wsworkcars.services.MarcaService

@RestController
class MarcaController {

    @Autowired
    lateinit var service: MarcaService

    @GetMapping("/marcas")
    fun getMarcas(): List<Marca> {
        return service.getData()
    }

    @GetMapping("/getMarca/{id}")
    fun getMarcaById(@PathVariable id: Long): Marca? {
        return service.getDataById(id)
    }

    @GetMapping("/getMarcaByName/{name}")
    fun getMarcaByName(@PathVariable name: String): Marca? {
        return service.getMarcaByName(name)
    }

    @PostMapping("/addMarca")
    fun addMarca(@RequestParam nome: String): Marca {
        return service.saveData(Marca(nome))
    }

    @PostMapping("/addMarcaJson")
    fun addMarcaJson(@RequestBody marca: Marca): Marca {
        return service.saveData(marca)
    }

    @PutMapping("/updateMarca")
    fun updateMarca(@RequestBody marca: Marca): Marca {
        return service.updateMarca(marca)
    }

    @DeleteMapping("/deleteMarca/{id}")
    fun deleteMarca(@PathVariable id: Long): HttpStatus {
        val result = service.deleteDataById(id)

        return if (result == null) HttpStatus.NOT_FOUND else HttpStatus.OK
    }

}